### ISSUE TITLE
Create abstraction over session management

### ISSUE THREAD

**Issue by chilversc** [apereo/phpCAS#42]
The session management needs an abstraction with an interface to allow it to be better integrated in to existing applications that have their own session management.

**Comment by ikari7789** [comment]
I'm curious if any work has been made on this in the past 3 years?

**Comment by jfritschi** [comment]
No, we have a lack of active developers. This is why this is simply a wishlist item at the moment.

**Comment by thetheodoros** [comment]
Is there an update on this issue?
Having custom session interfaces is essential in a microservice architecture context.
For instance, when using Docker containers you want your container to be ephemeral and have all user/session data handled in a dedicated service.

**Comment by jfritschi** [comment]
There has been no activity ... We are happy to take contributions 😄
